### PR TITLE
scalr-cli: init at 0.14.5

### LINKED ISSUE #222030: Package request: scalr-cli
**Project description**

CLI for interacting with Scalr.

**Metadata**

* homepage URL: https://docs.scalr.io/docs/overview-2
* source URL: https://github.com/Scalr/scalr-cli
* license: apache 2
* platforms: windows, linux, darwin

### PR DESCRIPTION
**Pull request by dylanmtaylor**
###### Description of changes

Add scalr-cli package I based this off of auth0-cli, another go package.. This is my first packaging attempt; please be gentle. This will close #222030.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dylanmtaylor** [comment]
bash-5.1# nix-prefetch-url --unpack https://github.com/Scalr/scalr-cli/archive/refs/tags/v0.14.5.zip
path is '/nix/store/l373f50z2wx0n3dl9rf3dq3ww22wrc4y-v0.14.5.zip'
0v5yyrymqxj8wf1fz7213vbfajhmag7s5g2ns9d0sqsfrl6dkxsz

I'm confused on how to get vendor hash.

**Comment by dylanmtaylor** [comment]
nix-prefetch-url --unpack https://github.com/Scalr/scalr-cli/archive/refs/tags/v0.14.5.zip | xargs nix hash to-sri --type sha256 --extra-experimental-features nix-command
path is '/nix/store/l373f50z2wx0n3dl9rf3dq3ww22wrc4y-v0.14.5.zip'
sha256-X/fZDM1OYw1a0la8os9TFUrl1h5BnO+C40h2XH32vmw=

**Comment by dylanmtaylor** [comment]
The tests want to create real resources in a scalr env. This isn't practical to build and run the tests without test infrastructure

**Comment by dylanmtaylor** [comment]
[nix-shell:/result/bin]# ./cli 

Usage: scalr [OPTION] COMMAND [FLAGS]

  'scalr' is a command-line interface tool that communicates directly with the Scalr API

Examples:
  $ scalr -help
  $ scalr -help get-workspaces
  $ scalr get-foo-bar -flag=value
  $ scalr -verbose create-foo-bar -flag=value -flag2=value2
  $ scalr create-foo-bar < json-blob.txt

Environment variables:
  SCALR_HOSTNAME  Scalr Hostname, i.e example.scalr.io
  SCALR_TOKEN     Scalr API Token
  SCALR_ACCOUNT   Default Scalr Account ID, i.e acc-tq8cgt2hu6hpfuj

Options:
  -version       Shows current version of this binary
  -help          Shows documentation for all (or specified) command(s)
  -verbose       Shows complete request and response communication data
  -configure     Run configuration wizard
  -update        Updates this tool to the latest version by downloading and replacing current binary
  -autocomplete  Enable shell tab auto-complete

**Comment by dylanmtaylor** [comment]
I am not sure how to rename the resultant binary file. It looks like there is a go option to do so.

**Comment by dylanmtaylor** [comment]
With substitueInPlace:

[nix-shell:/result/bin]# ./scalr -version
0.14.5

**Review by lilyinstarlight — APPROVED** [review]
This looks good to me now! I've no further review comments

---

Result of `nixpkgs-review pr 222132` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scalr-cli</li>
  </ul>
</details>

**Comment by dylanmtaylor** [comment]
@lilyinstarlight can I get a merge, please? It looks like I'd need write access to push this in myself.

**Review by NobbZ — CHANGES_REQUESTED** [review]
Can you please re-order the commits, such that adding you to the mainter-list comes first?

**Review by NobbZ — APPROVED** [review]
Sorry for the noise! GH web UI shows them in the wrong order. Locally the log looks fine.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4124,6 +4124,12 @@
     githubId = 6689924;
     name = "David Terry";
   };
+  dylanmtaylor = {
+    email = "dylan@dylanmtaylor.com";
+    github = "dylamtaylor";
```
Comment (zowoq) [inline maintainers/maintainer-list.nix:4129]: `dylamtaylor` -> `dylanmtaylor`

**Pull request merged**